app/components/files/file.tsx

```diff
@@ -66,74 +77,136 @@ const File = ({
         }
     }, [index]);
 
+    const handleOnOptionsPress = useCallback(() => {
+        if (onOptionsPress) {
+            onOptionsPress(index);
+        }
+    }, [index, onOptionsPress]);
+
+    const renderOptionsButton = useCallback(() => {
+        if (onOptionsPress) {
+            return (
+                <FileOptionsIcon
+                    onPress={handleOnOptionsPress}
+                />
+            );
+        }
+        return null;
+    }, [file, onOptionsPress, handleOnOptionsPress]);
+
+    const fileInfo = useMemo(() => {
+        return (
+            <FileInfo
+                file={file}
+                showDate={showDate}
+                channelName={channelName}
+                onPress={handlePreviewPress}
+                theme={theme}
+            />
+        );
+    }, [file, showDate, channelName, handlePreviewPress, theme]);
+
     const {styles, onGestureEvent, ref} = useGalleryItem(galleryIdentifier, index, handlePreviewPress);
 
-    if (isVideo(file) && publicLinkEnabled) {
+    const renderImageFileOverlay = useCallback(() => {
```

Comment:
If we get rid of the useMemos, this doesn't need to be a callback, since you are directly calling the function.